apps/api/src/domains/auth/auth.service.ts

```diff
@@ -233,35 +236,39 @@ export class AuthService {
     }
 
     const { accessTokenRequestURL, clientId, clientSecret } = oauthConfig;
-    try {
-      const { data } = await axios.post(
-        accessTokenRequestURL,
-        {
-          grant_type: 'authorization_code',
-          code,
-          redirect_uri: this.REDIRECT_URI,
-        },
-        {
-          headers: {
-            Authorization: `Basic ${Buffer.from(
-              clientId + ':' + clientSecret,
-            ).toString('base64')}`,
-            'Content-Type': 'application/x-www-form-urlencoded',
+    return await firstValueFrom(
```

Comment:
i think it's better to use lastValueFrom then firstValueFrom
https://rxjs.dev/deprecations/to-promise#lastvaluefrom